### PR TITLE
Ironsight and zooming 

### PR DESCRIPTION
**Pull request by Fueredoriku**
- Add zooming with fov change and fps gun changing grip
- Add d-pad support in menus, add zoom controls and stick down crouch
- Add telescope extension (improved zoom)
ZoomTelescope
- Third person zoom animation (IK)
zipZooms

### CODE REVIEW

**Review by toberge — CHANGES_REQUESTED** [review]
No zoomin' on the sniper (understandable have a good day)
image

Works fine for the most part, but I have some major concerns:
- When you kill another player while ironsighted, you forcibly zoom out w/o moving the gun, which looks really odd
- Scoope zoom feels a little too sensitive? The sensitivity is reasonable for the ironsights but maybe tune it down a little when scoped?
- Zoom is bound to the north button on gamepads and works as a toggle, while on mouse+keyboard holding RMB zooms. Toggle vs hold should be a preference eventually, but for now I'd at least like another gamepad bind: One of the bumpers should do the zoomin' as well, it doesn't feel natural to have _that_ on a face button (though I suspect assigning the left trigger to this might be more standard, but there's our precious crouching)

File: Assets/Scripts/Augment/AugmentImplementations/Telescope.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public class Telescope : GunExtension
+{
+    [SerializeField]
+    private float overrideZoomFov = 10f;
+    [SerializeField]
+    private float overrideZoomSpeed = 0.1f;
+
+    private GunController gunController;
+    private List<MeshRenderer> gunMeshes;
+    private List<SkinnedMeshRenderer> gunSkinMeshes;
```
[the remark this answers is not in the document]
Comment (Fueredoriku) [reply Assets/Scripts/Augment/AugmentImplementations/Telescope.cs:16]: The VFXs also inherit from Renderer, which would turn off things like trails and muzzleflashes when scoped.

File: Assets/Scripts/Control&Input/PlayerMovement.cs [hunk 2]
```diff
@@ -118,12 +127,21 @@ void Start()
...
+        inputManager.onZoomCanceled += OnZoomCanceled;
         localCameraHeight = inputManager.transform.localPosition.y;
-        gunHolder = GetComponent<PlayerManager>().GunHolder.gameObject;
+        localGunHolderX = gunHolder.transform.localPosition.x;
         localGunHolderHeight = gunHolder.transform.localPosition.y;
+        playerCamera = inputManager.GetComponent<Camera>();
+        startingFov = playerCamera.fieldOfView;
+
+        if (MatchController.Singleton)
+            MatchController.Singleton.onRoundEnd += ResetZoom;
```
[the remark this answers is not in the document]
Comment (Fueredoriku) [reply Assets/Scripts/Control&Input/PlayerMovement.cs:144]: I've made it consistent now 👍

File: Assets/Scripts/Augment/AugmentImplementations/Telescope.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (toberge) [reply Assets/Scripts/Augment/AugmentImplementations/Telescope.cs:16]: Oh. Ok nevermind :))))))

**Review by toberge — APPROVED** [review]
Seems a-okay now :+1: 

(except for the end-of-round shenanigans, but that can be resolved at a later time)

**Comment by Fueredoriku** [comment]
> (except for the end-of-round shenanigans, but that can be resolved at a later time)

I set up an issue to remind us later 👌 #383

**Pull request merged**